library/entropy_poll.c

```diff
@@ -104,10 +104,11 @@ static int getrandom_wrapper(void *buf, size_t buflen, unsigned int flags)
 #endif /* SYS_getrandom */
 #endif /* __linux__ || __midipix__ */
 
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NuttX__)
 #include <sys/param.h>
 #if (defined(__FreeBSD__) && __FreeBSD_version >= 1200000) || \
-    (defined(__DragonFly__) && __DragonFly_version >= 500700)
+    (defined(__DragonFly__) && __DragonFly_version >= 500700) || \
+    defined(__NuttX__)
```

Comment:
I'm not familiar with NuttX. Should there be a condition on the NuttX version, or on feature support? It looks like `getrandom` isn't a stable interface: it was [removed in 2020](https://github.com/apache/nuttx/pull/2488), then [added back in 2022](https://github.com/apache/nuttx/pull/5837).

Reply:
Thank you for your feedback with my commit. NuttX is an RTOS system for embedded devices where compatibility isn’t as crucial, meanwhile 'getrandom' is a feature that NuttX will continue to support in the future. so in my opinion that's no need to add a check for nuttx's version.